paddle/inference/io.cc

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "io.h"
+#include <fstream>
+
+namespace paddle {
+
+namespace io {
+
+bool IsParameter(const framework::VarDesc* var,
+                 const framework::ProgramDesc* main_program) {
```

Comment:
Just remind here that this function is introduced to make the current inference `ProgramDesc` run. Normally, there should not be any unreferenced variables. We should also check the existence of all parameters and give prompt information if there are some parameters absent. This can be done in next PR.

Reply:
Noted.